lib/handler/cache-handler.js

```diff
@@ -0,0 +1,202 @@
+'use strict'
+
+const util = require('../core/util.js')
+const DecoratorHandler = require('../handler/decorator-handler')
+const {
+  parseCacheControlHeader,
+  shouldRequestBeCached,
+  parseVaryHeader
+} = require('../util/cache.js')
+
+/**
+ * Takes a request, copies its data to a cache store, and passes it to the next
+ *  handler.
+ */
+class CacheHandler extends DecoratorHandler {
```

Comment:
Let's handle `onError`, if the request is aborted, we should not cache the response